roles/kubernetes/client/tasks/main.yml

```diff
@@ -61,11 +61,30 @@
     rm -rf {{ kube_config_dir }}/external_kubeconfig
   environment: "{{ proxy_env }}"
   run_once: yes
-  register: admin_kubeconfig
+  register: raw_admin_kubeconfig
+
+- name: Convert kubeconfig to YAML
+  set_fact:
+    admin_kubeconfig: "{{ raw_admin_kubeconfig.stdout | from_yaml }}"
+
+- name: Extract current settings from kubeconfig
```

Comment:
I see those facts only used in the following task. Could they be moved to the `vars:` section then?